### PR TITLE
Fetch mount location

### PR DESCRIPTION
**Pull request by shaik80**
### :nut_and_bolt: Description: What code changed, and why?
In this Pr we trying to get mount location using df -h

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-2234
### :athletic_shoe: How to Build and Test the Change
1. run `hab studio enter` locally
2. rebuild `components/automate-cli`
3. get bin from pkg to src `cp /hab/pkgs/smudassi/automate-cli/0.1.0/<time stamp>/bin/chef-automate .`
4. upload binary file into ec2 instance
5. make sure to setup nfs on ec2 instance
6. run `chef-automate verify serve`
7. open new terminal ssh into your instance
8. run the curl request
`curl --location --request POST 'http://localhost:7799/api/v1/fetch/nfs-mount-loc' --header 'Content-Type: application/json' --data-raw '{
  "mount_location": "/mnt/nfs"
}' | jq`

### :white_check_mark: Checklist

**All PRs** must tick these:

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [ ] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [ ] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [ ] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [ ] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [ ] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [ ] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*
### :camera: Screenshots, if applicable
**/api/v1/fetch/nfs-mount-loc** test
Screenshot 2023-06-02 at 3 55 39 PM

**/api/v1/checks/nfs-mount** test
Screenshot 2023-06-02 at 4 27 20 PM

Screenshot 2023-05-23 at 12 15 24 PM

Screenshot 2023-05-23 at 11 28 24 AM

### CODE REVIEW

**Review by vviveksharma — APPROVED** [review]

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go [hunk 1]
```diff
@@ -240,3 +258,37 @@ func createCheck(title string, passed bool, successMsg, errorMsg, resolutionMsg
...
+
+func (nm *NFSMountService) executeCommand() (string, error) {
+	output, err := nm.ExecuteShellCommandFunc(constants.MOUNT_CMD)
+	if err != nil {
+		return "", err
+	}
+	return string(output), nil
+}
+
+func (nm *NFSMountService) getMountDetails(output, mountLocation string) *models.NFSMountLocResponse {
```
Comment (punitmundra) [inline components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go:270]: check on all this os : amazon linux,ubuntu,redhat,centos,debian,suse 12

**Review by punitmundra — APPROVED** [review]

**Comment by punitmundra** [comment]
change looks fine, but rebase from main and test on the remaining os

**Review by SanjuPal01 — APPROVED** [review]
Rebase Branch with main

**Comment by shaik80** [comment]
ON suse 12

Screenshot 2023-05-25 at 4 46 35 PM

**Comment by shaik80** [comment]
on debian
Screenshot 2023-05-25 at 5 03 17 PM

**Comment by shaik80** [comment]
On ubuntu 
Screenshot 2023-05-26 at 3 18 54 PM

File: components/automate-cli/pkg/verifyserver/models/nfsmount.go [hunk 2]
```diff
@@ -20,7 +20,9 @@ type NFSMountLocRequest struct {
...
 
 type NFSMountLocResponse struct {
-	Address       string `json:"address"`
-	MountLocation string `json:"mount_location"`
-	Nfs           string `json:"nfs"`
+	Address            string `json:"address"`
+	MountLocation      string `json:"mount_location"`
+	Nfs                string `json:"nfs"`
+	StorageCapacity    string `json:"storage_capacity"`
+	AvailableFreeSpace string `json:"Available_free_space"`
```
Comment (vivek-yadav) [inline components/automate-cli/pkg/verifyserver/models/nfsmount.go:27]:
```suggestion
	AvailableFreeSpace string `json:"available_free_space"`
```

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go [hunk 3]
```diff
@@ -20,22 +21,25 @@ import (
 
 type INFSService interface {
```
Comment (vivek-yadav) [inline components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go:22]:
```suggestion
type NfsService interface {
```

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go [hunk 4]
```diff
@@ -20,22 +21,25 @@ import (
...
 	GetNFSMountDetails(models.NFSMountRequest) *[]models.NFSMountResponse
+	GetNFSMountLoc(req models.NFSMountLocRequest) *models.NFSMountLocResponse
 }
 
 type NFSMountService struct {
```
Comment (vivek-yadav) [inline components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go:27]:
```suggestion
type NfsServiceImp struct {
```

**Comment by shaik80** [comment]
output after updating my code from command (df -h) to pkg (github.com/shirou/gopsutil)
**Amazon Linux**
Amazon Linux
**suse 12**
suse 12
**centos 7**
centos 7
** debian**
debian
** redhat**
redhat
** ubuntu**
ubuntu

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice_test.go [hunk 5]
```diff
@@ -85,11 +87,19 @@ var (
...
 		Error: nil,
 	}
+	cmdRes = `Filesystem       Size  Used Avail Use% Mounted on
+/dev/root         49G  2.6G   46G   6% /
+tmpfs             16G     0   16G   0% /dev/shm
+tmpfs            6.3G  892K  6.3G   1% /run
+tmpfs            5.0M     0  5.0M   0% /run/lock
+/dev/xvda15      105M  6.1M   99M   6% /boot/efi
+172.31.26.202:/  8.0E     0  8.0E   0% /nfs
+tmpfs            3.2G  4.0K  3.2G   1% /run/user/1000`
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice_test.go:97]: is this needed anymore ?

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go [hunk 6]
```diff
@@ -240,3 +250,41 @@ func createCheck(title string, passed bool, successMsg, errorMsg, resolutionMsg
 		ResolutionMsg: resolutionMsg,
 	}
 }
+
+func (nm *NfsServiceImp) getMountDetails(mountLocation string) *models.NFSMountLocResponse {
+	// If we make disk partitions as false,
+	// we can't list mount info
+	diskPartition, err := nm.SystemResourceInfo.GetDiskPartitions(true)
+	if err != nil {
+		nm.log.Error("Error getting disk partions")
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go:259]: Log the err also

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go [hunk 7]
```diff
@@ -240,3 +250,41 @@ func createCheck(title string, passed bool, successMsg, errorMsg, resolutionMsg
...
+					Nfs:           "",
+					MountLocation: mountLocation,
+				}
+			}
+			return &models.NFSMountLocResponse{
+				Address:            strings.Split(partition.Device, ":")[0],
+				Nfs:                partition.Device,
+				MountLocation:      mountLocation,
+				StorageCapacity:    nm.SystemResourceInfo.FormatBytes(usageStat.Total),
+				AvailableFreeSpace: nm.SystemResourceInfo.FormatBytes(usageStat.Total),
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go:282]: Available free space is not same as total free space

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go [hunk 8]
```diff
@@ -240,3 +250,41 @@ func createCheck(title string, passed bool, successMsg, errorMsg, resolutionMsg
...
+			Address:       "",
+			Nfs:           "",
+			MountLocation: mountLocation,
+		}
+	}
+	for _, partition := range diskPartition {
+		if partition.Mountpoint == mountLocation {
+			usageStat, err := nm.SystemResourceInfo.GetDiskSpaceInfo(mountLocation)
+			if err != nil {
+				nm.log.Error("Failed to retrieve disk usage:")
```
Comment (vivekshankar1) [inline components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go:270]: log err here too

File: components/automate-cli/pkg/verifyserver/server/api/v1/nfsmount_test.go [hunk 9]
```diff
@@ -17,7 +17,13 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-func SetupNFSMountHandler(nm nfsmountservice.INFSService) (*fiber.App, error) {
+var (
+	nfsErrorBodyParser = "{\"status\":\"FAILED\",\"result\":null,\"error\":{\"code\":400,\"message\":\"Invalid Body Request\"}}"
+	errorMountLocEmpty = "{\"status\":\"FAILED\",\"result\":null,\"error\":{\"code\":400,\"message\":\"Mount Location cannot be empty\"}}"
+	successMsg         = "{\"status\":\"SUCCESS\",\"result\":{\"address\":\"\",\"mount_location\":\"/data\",\"nfs\":\"\",\"storage_capacity\":\"\",\"Available_free_space\":\"\"}}"
```
Comment (vipin230) [inline components/automate-cli/pkg/verifyserver/server/api/v1/nfsmount_test.go:23]: Available_free_space
```suggestion
	successMsg         = "{\"status\":\"SUCCESS\",\"result\":{\"address\":\"\",\"mount_location\":\"/data\",\"nfs\":\"\",\"storage_capacity\":\"\",\"available_free_space\":\"\"}}"
```

File: components/automate-cli/pkg/verifyserver/server/api/v1/nfsmount_test.go [hunk 10]
```diff
@@ -586,3 +599,56 @@ func TestNFSMount(t *testing.T) {
...
+			}`,
+		},
+		{
+			TestName:     "200:Success",
+			ExpectedCode: 200,
+			ExpectedBody: successMsg,
+			RequestBody: `{
+				"mount_location":"/data"
+			}`,
+		},
```
Comment (ArvinthC3000) [inline components/automate-cli/pkg/verifyserver/server/api/v1/nfsmount_test.go:634]: Add a test case for FS not found

File: components/automate-cli/pkg/verifyserver/server/routes.go [hunk 11]
```diff
@@ -19,7 +19,7 @@ func (vs *VerifyServer) SetupRoutes() {
 	apiChecksGroup.Post("/s3-config", vs.Handler.GetS3Config)
 	apiChecksGroup.Post("/port-reachable", vs.Handler.PortReachable)
 	apiChecksGroup.Post("/external-postgresql", vs.Handler.CheckExternalPostgresql)
-	
+
```
[the remark this answers is not in the document]
Comment (shaik80) [reply components/automate-cli/pkg/verifyserver/server/routes.go]: Done

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice_test.go (hunk 5 above)
Comment (shaik80) [reply components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice_test.go]: removed done

File: components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go (hunk 6 above)
Comment (shaik80) [reply components/automate-cli/pkg/verifyserver/services/nfsmountservice/nfsmountservice.go]: Done

**Review by vivekshankar1 — APPROVED** [review]

**Review by vivek-yadav — APPROVED** [review]

**Review by atultherajput — APPROVED** [review]

**Review by bvtejaswi — APPROVED** [review]

**Review by vipin230 — APPROVED** [review]

**Pull request merged**